ci-operator/step-registry/kubevirt/install/kubevirt-install-commands.sh

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env bash
+
+set -ex
+
+function ocp_version() {
+    oc get clusterversion version -o jsonpath='{.status.desired.version}' | awk -F "." '{print $1"."$2}'
+}
+
+if [[ ! "${CNV_SUBSCRIPTION_SOURCE}" =~ ^(cnv-prerelease-catalog-source|redhat-operators)$ ]]
+then
+    echo "CNV_SUBSCRIPTION_SOURCE environment variable value '${CNV_SUBSCRIPTION_SOURCE}' not allowed, allowed values are 'redhat-operators' or 'cnv-prerelease-catalog-source'"
+    exit 1
+fi
+
+
+# Get yq tool
+YQ="/tmp/yq"
+curl -L -o ${YQ} https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64
+chmod +x ${YQ}
+
+# Dynamically get CNV catalog image that was provided to the job via gangway API
+CNV_PRERELEASE_CATALOG_IMAGE=$(curl -s https://prow.ci.openshift.org/prowjob?prowjob="${PROW_JOB_ID}" |\
+  ${YQ} e '.spec.pod_spec.containers[0].env[] | select(.name == "CNV_PRERELEASE_CATALOG_IMAGE") | .value')
+
+if [ "${CNV_SUBSCRIPTION_SOURCE}" == "redhat-operators" ]
+  then
+  CNV_RELEASE_CHANNEL=stable
+elif [ -n "${CNV_PRERELEASE_CATALOG_IMAGE}" ]
+  then
+  CNV_RELEASE_CHANNEL=stable
+else
+  CNV_RELEASE_CHANNEL=nightly-$(ocp_version)
+  CNV_PRERELEASE_CATALOG_IMAGE=quay.io/openshift-cnv/nightly-catalog:$(ocp_version)
+fi
+
+# The kubevirt tests require wildcard routes to be allowed
+oc patch ingresscontroller -n openshift-ingress-operator default --type=json -p '[{ "op": "add", "path": "/spec/routeAdmission", "value": {wildcardPolicy: "WildcardsAllowed"}}]'
+
+# Make the masters schedulable so we have more capacity to run VMs
+oc patch scheduler cluster --type=json -p '[{ "op": "replace", "path": "/spec/mastersSchedulable", "value": true }]'
+
+if [ -n "${CNV_PRERELEASE_CATALOG_IMAGE}" ]
+  then
+  # Add pullsecret for cnv nightly channel from quay.io/openshift-cnv
+  QUAY_USERNAME=openshift-cnv+openshift_ci
+  QUAY_PASSWORD=$(cat /etc/cnv-nightly-pull-credentials/openshift_cnv_pullsecret)
```

Comment:
NIT: Since the script is ran with `-x` this pullsecret gets outputed to stdout. This is currently fine in CI as it takes care of [obfuscating it](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/57704/rehearse-57704-pull-ci-openshift-ovn-kubernetes-master-e2e-aws-ovn-virt-techpreview/1846154947213660160/artifacts/e2e-aws-ovn-virt-techpreview/kubevirt-install/build-log.txt) but I think it would be best to avoid outputting it in the first place, can be done as a followup.